allennlp/modules/stacked_elmo_lstm.py

```diff
@@ -0,0 +1,146 @@
+                                          recurrent_dropout_probability,
+                                          memory_cell_clip_value,
+                                          state_projection_clip_value)
+            lstm_input_size = hidden_size
+
+            self.add_module('forward_layer_{}'.format(layer_index), forward_layer)
+            self.add_module('backward_layer_{}'.format(layer_index), backward_layer)
+            forward_layers.append(forward_layer)
+            backward_layers.append(backward_layer)
+        self.forward_layers = forward_layers
+        self.backward_layers = backward_layers
+
+    def forward(self,  # pylint: disable=arguments-differ
+                inputs: PackedSequence,
+                initial_state: Optional[Tuple[torch.Tensor, torch.Tensor]] = None):
+        """
+        Parameters
+        ----------
+        inputs : ``PackedSequence``, required.
+            A batch first ``PackedSequence`` to run the stacked LSTM over.
+        initial_state : Tuple[torch.Tensor, torch.Tensor], optional, (default = None)
+            A tuple (state, memory) representing the initial hidden state and memory
+            of the LSTM. Each tensor has shape (1, batch_size, output_dimension).
+
+        Returns
+        -------
+        output_sequence : ``torch.FloatTensor``
+            The encoded sequence of shape (num_layers, batch_size, sequence_length, hidden_size)
+        final_states: ``Tuple[torch.FloatTensor, torch.FloatTensor]``
+            The per-layer final (state, memory) states of the LSTM, with shape
+            (num_layers, batch_size, 2 * hidden_size) and  (num_layers, batch_size, 2 * cell_size)
+            respectively.
+        """
+        if not initial_state:
+            hidden_states = [None] * len(self.forward_layers)
+        elif initial_state[0].size()[0] != len(self.forward_layers):
+            raise ConfigurationError("Initial states were passed to forward() but the number of "
+                                     "initial states does not match the number of layers.")
+        else:
+            hidden_states = list(zip(initial_state[0].split(1, 0),
+                                     initial_state[1].split(1, 0)))
+
+        inputs, batch_lengths = pad_packed_sequence(inputs, batch_first=True)
+        forward_output_sequence = inputs
+        backward_output_sequence = inputs
+
+        final_states = []
+        sequence_outputs = []
+        for layer_index, (forward_layer, backward_layer, state) in enumerate(zip(self.forward_layers,
+                                                                                 self.backward_layers,
+                                                                                 hidden_states)):
+            forward_cache = forward_output_sequence
+            backward_cache = backward_output_sequence
+
+            forward_output_sequence, forward_state = forward_layer(forward_output_sequence,
```

Comment:
I don't think this logic will work.  The forward and backward LSTMs have different states, but here you are passing the same state to both directions.

Reply:
You need to also unpack the forward and backward directions.
At each iteration of the loop, `state` is tuple with shape `(batch_size, 2 * dim)` (for each entry).
So you need something like
```
forward_hidden_state, backward_hidden_state = state[0].split(self.hidden_size, 1)
forward_memory_state, backward_memory_state = state[0].split(self.cell_size, 1)
forward_state = (forward_hidden_state, forward_memory_state)
backward_state = (backward_hidden_state, backward_memory_state)
```